x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -606,4 +607,45 @@ public PlanFactory visitJoinCommand(EsqlBaseParser.JoinCommandContext ctx) {
             return new LookupJoin(source, p, right, joinFields);
         };
     }
+
+    @Override
+    public PlanFactory visitForkCommand(EsqlBaseParser.ForkCommandContext ctx) {
+        List<PlanFactory> subQueries = visitForkSubQueries(ctx.forkSubQueries());
+        if (subQueries.size() < 2) {
+            throw new ParsingException(source(ctx), "Fork requires at least two branches");
+        }
+        return input -> {
+            List<LogicalPlan> subPlans = subQueries.stream().map(planFactory -> planFactory.apply(input)).toList();
+            return new Fork(source(ctx), input, subPlans);
+        };
+    }
+
+    @Override
+    public List<PlanFactory> visitForkSubQueries(EsqlBaseParser.ForkSubQueriesContext ctx) {
+        var subQueriesCtx = ctx.forkSubQuery();
+        ArrayList<PlanFactory> list = new ArrayList<>();
+        for (var subQueryCtx : subQueriesCtx) {
+            list.add(visitForkSubQuery(subQueryCtx));
+        }
+        return List.copyOf(list);
```

Comment:
Move the code from the analyzer here so each subquery has an eval with its index attached to it at the end.

Reply:
Good idea. Done.